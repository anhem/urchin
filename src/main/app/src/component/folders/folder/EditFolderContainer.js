import React, {Component} from 'react'
import {connect} from 'react-redux'
import EditFolder from './EditFolder';
import {deleteFolder, getFolder, setFolder} from '../../../action/folderAction';

export class EditFolderContainer extends Component {

    componentWillMount() {
        let folderId = parseInt(this.props.match.params.id, 10);
        this.props.getFolder(folderId);
    }

    render() {
        return (
            <EditFolder
                folder={this.props.folder}
                callbacks={{
                    setFolder: this.props.setFolder,
                    deleteFolder: this.props.deleteFolder,
                }}
            />
        )
    }
}

const mapDispatchToProps = (dispatch) => {
    return {
        getFolder: (folderId) => {
            dispatch(getFolder(folderId))
        },
        setFolder: (folder) => {
            dispatch(setFolder(folder))
        },
        deleteFolder: (folderId) => {
            dispatch(deleteFolder(folderId))
        }
    }
};

const mapStateToProps = (state) => {
    return {
        folder: state.folderData.folder || {},
        usersInFolder: state.folderData.usersInFolder || [],
        users: state.userData.users || []
    }
};

export default connect(mapStateToProps, mapDispatchToProps)(EditFolderContainer)