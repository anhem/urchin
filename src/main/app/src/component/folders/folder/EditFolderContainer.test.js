import React from 'react';
import {mount, shallow} from 'enzyme'
import toJson from 'enzyme-to-json';
import {EditFolderContainer} from "./EditFolderContainer";

describe('EditFolderContainer', () => {

    let props = {
        match: {
            params: {
                id: 0
            }
        },
        folder: {},
        usersInFolder: [],
        users: [],
        getFolder: jest.fn(),
        getUsersForFolder: jest.fn(),
        getUsers: jest.fn(),
    };

    it('renders without crashing', () => {
        expect(mount(<EditFolderContainer {...props}/>).length).toEqual(1);
    });

    it('match snapshot', () => {
        expect(toJson(shallow(<EditFolderContainer {...props}/>))).toMatchSnapshot();
    });
});
