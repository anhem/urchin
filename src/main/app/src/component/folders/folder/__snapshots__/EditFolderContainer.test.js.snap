// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditFolderContainer match snapshot 1`] = `
<EditFolder
  callbacks={
    Object {
      "deleteFolder": undefined,
      "setFolder": undefined,
    }
  }
  folder={Object {}}
/>
`;
